syntax = "proto3";

package bom.task_actor_type;

option java_package = "bom";
option java_outer_classname="TaskActorTypeProto";

/// Type of entity involved in an interaction with a task.
// For example, this type can be used to represent the entity
// responsible for iniatiting the running of a task, the
// entity/system reporting a change in task state or the type
// of entity responsible for imposing/enforcing resource limits.
enum TaskActorType {
  UNSET = 0;
  UNKNOWN = 1;

  USER = 2;
  ADMIN = 3;
  THIS_TASK = 4; /// The task itself
  OTHER_TASK = 5; /// Another task, managed in a similar way to this task
  /// Host system could be a physical machine, VM, container etc
  HOST_SYSTEM = 6;
  BATCH_SCHEDULER = 7;
  WORKFLOW_SCHEDULER = 8;
  ORCHESTRATION_SYSTEM = 9;
}
